windows-installer/openssl.nsi

```diff
@@ -0,0 +1,113 @@
+
+######################################################
+# NSIS windows installer script file
+# Requirements: NSIS 3.0 must be installed with the MUI plugin
+# Usage notes:
+# This script expects to be executed from the directory it is
+# currently stored in.  It expects a 32 bit and 64 bit windows openssl
+# build to be present in the ..\${BUILD32} and ..\${BUILD64} directories
+# respectively
+# ####################################################
+
+!include "MUI.nsh"
+
+!define PRODUCT_NAME "OpenSSL"
+
+# The name of the output file we create when building this
+# NOTE version is passed with the /D option on the command line
+OutFile "openssl-${VERSION}-installer.exe"
+
+# The name that will appear in the installer title bar
+NAME "${PRODUCT_NAME} ${VERSION}"
+
+ShowInstDetails show
+
+Function .onInit
+	StrCpy $INSTDIR "C:\Program Files\openssl-${VERSION}"
+FunctionEnd
+
+# This section is run if installation of 32 bit binaries are selected
+
+!ifdef BUILD64
+# This section is run if installation of the 64 bit binaries are selectd
+SectionGroup "64 Bit Installation"
+	Section "64 Bit Binaries"
+		SetOutPath $INSTDIR\x64\lib
+		File /r "${BUILD64}\Program Files\OpenSSL\lib\"
+		SetOutPath $INSTDIR\x64\bin
+		File /r "${BUILD64}\Program Files\OpenSSL\bin\"
+		SetOutPath "$INSTDIR\x64\Common Files"
+		File /r "${BUILD64}\Program Files\Common Files\"
+	SectionEnd
+	Section "x64 Development Headers"
+		SetOutPath $INSTDIR\x64\include
+		File /r "${BUILD64}\Program Files\OpenSSL\include\"
+	SectionEnd
+SectionGroupEnd
+!endif
+
+!ifdef BUILD32
+# This section is run if installation of the 64 bit binaries are selectd
+SectionGroup "32 Bit Installation"
+	Section "32 Bit Binaries"
+		SetOutPath $INSTDIR\x64\lib

```

Comment:
Hmmm...this is for 32 bit, but the path has x64 in it. Is that correct?

Reply:
yup, that should all be x32, good catch!